tools/building/createTest.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+import pathlib
+
+
+def is_precice_root(dir):
+    detect = [
+        "CHANGELOG.md", "CMakeLists.txt", "LICENSE", "src", "tests", "cmake"
+    ]
+    return all(map(lambda c: os.path.exists(os.path.join(dir, c)), detect))
+
+
+def find_precice_root():
+    search_depth = 4
+    current = pathlib.Path(os.path.curdir).absolute()
+    candidats = [current] + list(current.parents)[:search_depth]
+    for dir in candidats:
+        if is_precice_root(dir):
+            return dir
+    raise "Unable to find the root directory of precice"
+
+
+def patharg(arg):
+    assert (arg)
+    if " " in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain spaces.".format(arg))
+    if "." in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain the file extensions.".
+            format(arg))
+    if not arg[0].isupper():
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" must use CamelCase.".format(arg))
```

Comment:
```suggestion
            "The given suite name \"{}\" must use CamelCase.".format(arg))
```
Briefly discussed with @uekerman today: Usually out folders are `cplscheme`, so one word in lower case. I would propose the following folder structure:

```
serial/multi-coupling-four-solvers/another-suite/Dummy.cpp
```

However, the suite and test should be `CamelCase`:
```
BOOST_AUTO_TEST_SUITE(PreciceTests)
BOOST_AUTO_TEST_SUITE(Serial)
BOOST_AUTO_TEST_SUITE(MultiCouplingFourSolvers)
BOOST_AUTO_TEST_SUITE(AnotherSuite)
BOOST_AUTO_TEST_CASE(Dummy)
...
```

Reply:
To give a reason why I would prefer: `serial/multi-coupling-four-solvers/another-suite/Dummy.cpp`

* Consistent with our current single-word lower case folders in `src`
* Inspired by the multiple-word config files in the tests (e.g. `multi-coupling-four-solver-1.xml`) and `precice-adapter` repositories

Reply:
This will decouple the test path from the test suites though.
Meaning, the automatic generation of the config path will not work.

As alternative, we could use a lowercase name for all test suites. This would then be inconsistent with the test names though.

This is tricky.

Reply:
> Meaning, the automatic generation of the config path will not work.

In this case I would drop this feature. I think automatic generation of the config path is nice, but we also do not have it currently and if we don't have a solution I guess we have to live without it. We can still use a fixture to provide most of the path:

https://github.com/precice/precice/blob/f8fb5c4364c8965799d22c2a108e3da961a4bd4e/src/precice/tests/ParallelTests.cpp#L34

Reply:
I updated the generator accordingly in f5722f6. We now get 
```
precice/tests$  python3 ../tools/building/createTest.py CouplingScheme WhySoManyTests ThisIsTheTest
Create directory /home/benjamin/Programming/precice/tests/coupling-scheme/why-so-many-tests
Create test source ThisIsTheTest.cpp
Create test config this-is-the-test.xml
```

Reply:
We decided to use `CamelCase.xml` for the configs, not `kebab-case.xml` for the sake of automatically generated test paths. This is implemented (afaik) in 4d040cf. I will mark this conversation as resolved.